src/modeler/AutoRest.Swagger/Validation/ServiceDefinitionParameters.cs

```diff
@@ -33,7 +37,27 @@ public class ServiceDefinitionParameters : TypedRule<Dictionary<string, SwaggerP
         /// <summary>
         /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
         /// </summary>
-        public override Category Severity => Category.Warning;
+        public override Category Severity => Category.Error;
+
+        /// <summary>
+        /// This rule passes if the parameters section contains both subscriptionId and api-version parameters
+        /// </summary>
+        /// <param name="ParametersMap">Dictionary of swagger parameters</param>
+        /// <param name="context">Rule context</param>
+        /// <param name="formatParameters">Formatted parameters</param>
+        /// <returns></returns>
+        public override bool IsValid(Dictionary<string, SwaggerParameter> ParametersMap, RuleContext context, out object[] formatParameters)
+        {
+            formatParameters = new object[0];
+            ServiceDefinition serviceDefinition = context.Root;
+            
+            // Rule does not apply if x-ms-parameterized-host extension is not applied
+            if (!serviceDefinition.Extensions.ContainsKey(XMsParameterizedHostExtensionName))
+            {
+                return true;
+            }
 
+            return true == (ParametersMap?.Values.Any(parameter => parameter.Name == SubscriptionId || parameter.Name == ApiVersion));
```

Comment:
I think this should be case-sensitive. That'd also facilitate us towards common definitions. Ref: https://github.com/Azure/azure-rest-api-specs-pr/pull/40/files#diff-d3a0dcdc5d2939703fa9e32dff2295c1R249